packages/design-system/src/components/NativeDialog/NativeDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useRef, useEffect, useLayoutEffect } from 'react';
+import dialogPolyfill from 'dialog-polyfill';
+
+interface NativeDialogProps {
+  children: React.ReactNode;
+  /**
+   * Function called to close dialog.
+   */
+  exit: any;
+  /**
+   * Determines which dialog method is called.
+   * If true, `showModal()` is called, if false,
+   * `show()` is called.
+   */
+  showModal?: boolean;
+  /*
+   * @hide-prop Spreading props
+   */
+  [x: string]: any;
+}
+
+const NativeDialog = ({ children, exit, showModal, ...dialogProps }: NativeDialogProps) => {
+  const dialogRef = useRef(null);
+
+  // useLayoutEffect(() => {
+  //   if ((window as any).HTMLDialogElement === undefined) {
+  //     dialogPolyfill.registerDialog(dialogRef.current);
+  //   }
+  // })
+
+  useEffect(() => {
+    const dialogNode = dialogRef.current;
+    showModal ? dialogNode.showModal() : dialogNode.show();
```

Comment:
`showModal` will call a method that makes an `open` attribute appear dynamically. Depending on which method gets called, you either get a focus trap or you don't.